Detected [E0121] in struct

<!--
Hello and thank you for the issue!
If you would like to report a bug, we have added some points below that you can fill out.
Feel free to remove all the irrelevant text to request a new feature.
-->

## Environment

* **IntelliJ Rust plugin version:** 0.4.154.4041-212-nightly
* **Rust toolchain version:** 1.53.0 (53cb7b09b 2021-06-17) x86_64-apple-darwin
* **IDE name and version:** CLion 2021.2 (CL-212.4746.93)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new

## Problem description

error[E0121]: the type placeholder `_` is not allowed within types on item signatures for **structs**
But IDE doesn't detect it:
<img width="270" alt="изображение" src="https://user-images.githubusercontent.com/49211026/129342316-b9cd6805-ee27-4f72-8c24-a77e8a93f6f8.png">

## Steps to reproduce
```rust
struct Baz(i32, _); //error
fn foo(a: i32, _){  //same error detected
    todo!()
}
```


<!--
Please include as much of your codebase as needed to reproduce the error.
If the relevant files are large, please provide a link to a public repository or a [Gist](https://gist.github.com/).
-->

--- a/src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt
+++ b/src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt
@@ -540,8 +540,12 @@ class RsErrorAnnotator : AnnotatorBase(), HighlightRangeExtension {
     private fun checkBaseType(holder: RsAnnotationHolder, type: RsBaseType) {
         if (type.underscore == null) return
         val owner = type.owner.parent
-        if ((owner is RsValueParameter && owner.parent.parent is RsFunction)
-            || (owner is RsRetType && owner.parent is RsFunction) || owner is RsConstant
+        val ownerParent = owner.parent
+        val ownerGrandParent = ownerParent.parent
+        if ((owner is RsValueParameter && ownerGrandParent is RsFunction)
+            || (owner is RsRetType && ownerParent is RsFunction)
+            || owner is RsConstant
+            || (owner is RsFieldDecl && (ownerGrandParent is RsStructItem || ownerGrandParent is RsEnumVariant))
         ) {
             RsDiagnostic.TypePlaceholderForbiddenError(type).addToHolder(holder)
         }

--- a/src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt
+++ b/src/test/kotlin/org/rust/ide/annotator/RsErrorAnnotatorTest.kt
@@ -420,6 +420,14 @@ class RsErrorAnnotatorTest : RsAnnotatorTestBase(RsErrorAnnotator::class) {
         fn bar() -> <error>_</error> {}
         fn baz(t: (u32, <error>_</error>)) -> (bool, (f64, <error>_</error>)) { unreachable!() }
         static FOO: <error>_</error> = 42;
+        struct Baz(i32, <error>_</error>);
+        struct S {
+            a: <error>_</error>
+        }
+        enum E {
+            V1(<error>_</error>),
+            V2 { a: <error>_</error> }
+        }
     """)
 
     fun `test name duplication in struct E0124`() = checkErrors("""

